src/app/Layout/Layout.tsx

```diff
@@ -0,0 +1,74 @@
+"use client"
+import React, {  useState } from 'react'
+import {
+    MagnifyingGlassIcon,
+    HamburgerMenuIcon,
+} from '@radix-ui/react-icons'
+import { Input } from "@/components/ui/input"
+import { Button } from "@/components/ui/button"
+import { Sidebar } from '../components/Sidebar'
+import { usePathname } from 'next/navigation'
+import { RoutesTitle } from '@/lib/routes'
+
+type UserRole = "Incharge" | "member" | "Shift-Incharge";
+
+export function Layout({ children }: { children: React.ReactNode, }) {
+    const pathname = usePathname()
+
+    const [sidebarOpen, setSidebarOpen] = useState(false)
+    const [isTilleHide, setIsTitleHide] = useState(false)
+    const toggleSidebar = () => setSidebarOpen(!sidebarOpen)
+    const [userRole, setUserRole] = useState<UserRole | null>(null);
+
+    return (
+        <div className="bg-gray-100 flex ">
+            {
+                userRole &&
+                <Sidebar
+                    sidebarOpen={sidebarOpen}
+                    toggleSidebar={toggleSidebar}
```

Comment:
Callbacks to components should always be prefixed with `on` keyword. This should be `onToggleSidebar`